### ISSUE TITLE
JavaScript files incorrectly reported as Java

### ISSUE THREAD

**Issue by timvandermeij** [github/linguist#2332]
Take a look at the language statistics for https://github.com/mozilla/pdf.js. It says "Java 85.9%", but in fact there is not a single line of Java in the repository: it's all JavaScript. Interestingly, if you click that percentage, you end up at https://github.com/mozilla/pdf.js/search?l=java, which shows only six files: not by far 85.9%.

This bug is new and occurred approx. a few weeks ago. A friendly GitHub staff member asked me to create an issue for this here and pointed out that Linguist might be confused by Mozilla's way of tagging files for various editors. See https://github.com/mozilla/pdf.js/blob/ba8a4f2b3e2ac3f0eb9c4ab59eedb9e0072ad52b/test/driver.js#L1 for example: the first line of all our JavaScript files starts with 'Mode: Java'.

**Comment by arfon** [comment]
Hi @timvandermeij - yes, I'm afraid that's exactly the issue. We use a number of strategies to identify the language of a file and the modelines is the first: https://github.com/github/linguist/blob/master/lib/linguist/language.rb#L90-L96

I would suggest adding some custom overrides to force Linguist to switch back to `JavaScript` reporting: https://github.com/github/linguist#overrides

**Comment by arfon** [comment]
Update: I've opened a PR with the fix here: https://github.com/mozilla/pdf.js/pull/5936

**Comment by pchaigno** [comment]
@timvandermeij Just out of curiosity, why are you tagging JS files as Java in Emacs?

**Comment by timvandermeij** [comment]
Thank you for providing a fix so quickly!

@pchaigno I am not sure, but I think this might be unnecessary with modern editors. I think this was added some time ago to make editors use better indentation/syntax highlighting. I found it a bit strange too to be honest, but it only recently became a problem when Linguist changed the heuristics for determining the language.

Anyway, it is fixed for us now and perhaps we will change that Java tagging later also in our repository.
